src/serac/physics/utilities/functional/domain_integral_kernels.hpp

```diff
@@ -0,0 +1,273 @@
+#pragma once
+
+#include "serac/physics/utilities/quadrature_data.hpp"
+#include "serac/physics/utilities/functional/integral_utilities.hpp"
+
+namespace serac {
+
+namespace domain_integral {
+
+/**
+ * @brief The base kernel template used to create different finite element calculation routines
+ *
+ * @tparam test The type of the test function space
+ * @tparam trial The type of the trial function space
+ * The above spaces can be any combination of {H1, Hcurl, Hdiv (TODO), L2 (TODO)}
+ *
+ * Template parameters other than the test and trial spaces are used for customization + optimization
+ * and are erased through the @p std::function members of @p DomainIntegral
+ * @tparam g The shape of the element (only quadrilateral and hexahedron are supported at present)
```

Comment:
You are missing `test` here in your `tparam` list and have an extra `qpt_data_type`.

You have an extra `data` in `param`

Also can you please order these to match the actual order.

Reply:
😆 I noticed this as well in some stuff already in `develop`, but it actually meant less copying and pasting for me in #518